pymc/util.py

```diff
@@ -607,6 +635,314 @@ def update(
+        else:
+            self.complete_style = self.non_diverging_style
+            self.finished_style = self.non_diverging_finished_style
+
+
+class ProgressManager:
+    """Manage progress bars displayed during sampling."""
+
+    def __init__(
+        self,
+        step_method: "BlockedStep | CompoundStep",
+        chains: int,
+        draws: int,
+        tune: int,
+        progressbar: bool | ProgressType = True,
+        progressbar_theme: Theme | None = None,
+    ):
+        """
+        Manage progress bars displayed during sampling.
+
+        When sampling, Step classes are responsible for computing and exposing statistics that can be reported on
+        progress bars. Each Step implements two class methods: :meth:`pymc.step_methods.BlockedStep._progressbar_config`
+        and :meth:`pymc.step_methods.BlockedStep._make_update_stats_function`. `_progressbar_config` reports which
+        columns should be displayed on the progress bar, and `_make_update_stats_function` computes the statistics
+        that will be displayed on the progress bar.
+
+        Parameters
+        ----------
+        step_method: BlockedStep or CompoundStep
+            The step method being used to sample
+        chains: int
+            Number of chains being sampled
+        draws: int
+            Number of draws per chain
+        tune: int
+            Number of tuning steps per chain
+        progressbar: bool or ProgressType, optional
+            How and whether to display the progress bar. If False, no progress bar is displayed. Otherwise, you can ask
+            for one of the following:
+            - "combined": A single progress bar that displays the total progress across all chains. Only timing
+                information is shown.
+            - "split": A separate progress bar for each chain. Only timing information is shown.
+            - "combined+stats" or "stats+combined": A single progress bar displaying the total progress across all
+                chains. Aggregate sample statistics are also displayed.
+            - "split+stats" or "stats+split": A separate progress bar for each chain. Sample statistics for each chain
+                are also displayed.
+
+            If True, the default is "split+stats" is used.
+
+        progressbar_theme: Theme, optional
+            The theme to use for the progress bar. Defaults to the default theme.
+        """
+        if progressbar_theme is None:
+            progressbar_theme = default_progress_theme
+
+        self.combined_progress = False
+        self.full_stats = True
+        show_progress = True
+
+        match progressbar:
+            case True:
+                show_progress = True
+            case False:
+                show_progress = False
+            case "combined":
+                self.combined_progress = True
+                self.full_stats = False
+            case "split":
+                self.combined_progress = False
```

Comment:
Nit: It's a bit arbitrary that in some match cases you redefine defaults and others not

Reply:
only the boolean cases don't redefine. Would you prefer everything is defined in every case? No objection, I was just trying to save some lines

Reply:
Either be verbose or save all you can then, don't redefine any "defaults"

Reply:
You'd make a poor King Solomon